Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -924,6 +924,51 @@ class SwiftObjectServerTests: SwiftSyncTestCase {
         waitForExpectations(timeout: 10.0, handler: nil)
     }
 
+    func testDeleteUser() {
+        let email = "realm_tests_do_autoverify\(randomString(7))@\(randomString(7)).com"
+        let password = randomString(10)
+
+        let registerUserEx = expectation(description: "Register user")
+
+        app.emailPasswordAuth.registerUser(email: email, password: password) { (error) in
+            XCTAssertNil(error)
+            registerUserEx.fulfill()
+        }
+        wait(for: [registerUserEx], timeout: 4.0)
+
+        let loginEx = expectation(description: "Login user")
+        var syncUser: User?
+
+        app.login(credentials: Credentials.emailPassword(email: email, password: password)) { result in
+            switch result {
+            case .success(let user):
+                syncUser = user
+            case .failure:
+                XCTFail("Should login user")
+            }
+            loginEx.fulfill()
+        }
+
+        wait(for: [loginEx], timeout: 4.0)
+
+        XCTAssertEqual(syncUser?.id, app.currentUser?.id)
+        XCTAssertEqual(app.allUsers.count, 1)
+
+        let deleteEx = expectation(description: "Delete user")
+
+        XCTAssertNotNil(syncUser)
+
+        syncUser?.delete { (error) in
+            XCTAssertNil(error)
+            deleteEx.fulfill()
+        }
+
+        wait(for: [deleteEx], timeout: 4.0)
+
+        XCTAssertNil(app.currentUser)
+        XCTAssertEqual(app.allUsers.count, 0)
```

Comment:
It would be nice if this checked the server for deletion: https://docs.mongodb.com/realm/admin/api/v3/. Should be fairly straightforward to enable from `RealmServer`.